packages/components/src/DateTimePickers/pickers/IncrementableScrollList/IncrementableScrollList.component.js

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { AutoSizer, List } from 'react-virtualized';
+import classNames from 'classnames';
+import theme from './IncrementableScrollList.scss';
+import IconButton from '../../shared/components/IconButton';
+import RowRenderer from './RowRenderer';
+
+function keepInBoundaries(number, min, max) {
+	if (number < min) {
+		return min;
+	}
+	if (number > max) {
+		return max;
+	}
+	return number;
+}
+
+const NB_ITEMS_DISPLAYED = 5;
+
+class IncrementableScrollList extends React.Component {
+	constructor(props) {
+		super(props);
+		const firstIndex = 0;
+		const lastIndex = this.props.items.length - NB_ITEMS_DISPLAYED;
+
+		const startIndex = Math.max(this.props.initialIndex - Math.floor(NB_ITEMS_DISPLAYED / 2), 0);
+		this.scrollToIndex = keepInBoundaries(startIndex, firstIndex, lastIndex);
+
+		this.state = {
+			startIndex: this.scrollToIndex,
+		};
+
+		this.scrollUp = this.scrollRows.bind(this, -5);
+		this.scrollDown = this.scrollRows.bind(this, 5);
+		this.onRowsRendered = this.onRowsRendered.bind(this);
+		this.setListRef = this.setListRef.bind(this);
+	}
+
+	onRowsRendered(data) {
+		const { startIndex } = data;
```

Comment:
Useless constant.

Reply:
I'd rather see the last one being used.
It removes an allocation.

Reply:
done it